lightning/src/ln/channelmanager.rs

```diff
@@ -8589,6 +8626,30 @@ where
 						}
 					}
 				}
+
+				// Whether the downstream channel was closed or not, try to re-apply any payment
+				// preimages from it which may be needed in upstream channels for forwarded
+				// payments.
+				let outbound_claimed_htlcs_iter = monitor.get_all_current_outbound_htlcs()
+					.into_iter()
+					.filter_map(|(htlc_source, (htlc, preimage_opt))| {
+						if let HTLCSource::PreviousHopData(_) = htlc_source {
+							if let Some(payment_preimage) = preimage_opt {
+								Some((htlc_source, payment_preimage, htlc.amount_msat,
+									counterparty_opt.is_none(), // i.e. the downstream chan is closed
```

Comment:
Is this the most reliable way we check whether the channel is closed? Could we expand on the comment a tad to give reasoning why this implies the channel is closed.

Like it makes sense since there's no peer corresponding to that channel ID anymore, and we remove from that map when closing, but at quick glance it seems arb here.